### DISCUSSION TITLE
Case-Insensitive property match for Deserialization

### DISCUSSION THREAD

**Discussion by gimlichael** [aaubry/YamlDotNet#946] [Q&A] [answered]
Hello,

I am currently hit by the fact, that the deserializer expects 1:1 casing from YAML to class in .NET.
Is there a way to ignore this (without using attributes)?
I really dont want to "decorate" my DDD classes with attributes, so any hints in this regards is most appreciated.

Example:

```yaml
---
allowedPrefixes:
  - A
  - AB
  - ABC
allowedSuffixes:
  - XYZ
```

In this case, where my class has PascalCase convention, it fails hard, and expect to have camelCase.
It works if I change the class to have camelCase (but that is just violating so many idioms and conventions that my OCD go bananas) :-)

I believe this should be a matter of preference. Ideally, it matches a property ignoring any casing.

I have tried using WithNamingConvention but this does not help.

In advance thanks.

**Comment by EdwardCooke** [comment, marked as answer]
Yes. I just added it in the latest version. On your deserializerbuilder use the `WithCaseInsensitivePropertyMatching` method to ignore casing.

**Reply by gimlichael** [reply]
Ah, cool. Problem is, that you have quite a number of breaking changes, so I have to await that bit. I will try to look in your source code to see for a workaround prior to v16.
